set.go

```diff
@@ -140,3 +140,49 @@ func SetInt(node *goyaml.Node, path Path, value int64) (valueNode *goyaml.Node,
 	})
 	return
 }
+
+// SetScalar sets a scalar value at the given path within the given node.
+//
+// node is the node to set the value within.
+//
+// path is the path within the node to set the value at.
+//
+// value is the scalar value to set, eg. "hello".
+//
+// Returns a bool showing whether or not the value set updated an existing value (ie. value was changed).
+func SetScalar(node *goyaml.Node, path Path, value any) (changed bool, err error) {
+	n, err := plumbing.MakeScalar(value)
+	if err == nil {
+		_, changed, err = set(node, path, nil, func() *goyaml.Node {
+			return n
+		})
+	}
+
+	return
+}
+
+// SetScalar sets a node at the given path within the given node.
+//
+// node is the node to set the value within.
+//
+// path is the path within the node to set the value at.
+//
+// value is the node to set at the specified path.
+//
+// Returns the node that was added.
+//
+// Example, adding a new empty mapping node to "node.x.y":
+//
+//	value := &goyaml.Node{
+//		Kind: goyaml.MappingNode,
+//		Tag:  "!!map",
+//	}
+//
+// err := Set(node, ".x.y", value)
+// ...
+func Set(node *goyaml.Node, path Path, value *goyaml.Node) (addedNode *goyaml.Node, err error) {
+	addedNode, _, err = set(node, path, nil, func() *goyaml.Node {
```

Comment:
We have to refactor `set` slightly as it hardcodes comparing an existing value with `null` now, and skips setting if existing value is `null`.